### DISCUSSION TITLE
Nested routers doesn't include dynamic path fields added from the parent router

### DISCUSSION THREAD

**Discussion by SDAravind** [fastapi/fastapi#11426] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Description

Here is the code that mimics the actual code. I have main application and a router (subpackage1) and within the router module I have another router (subpackage2). I have declared a path variable in subpackage1 in the views.py. I would like to access this path variable in subpackage2 in the views.py.

How do I access `account` which is path variable declared in subpackage1 views.py module from subpackage2 views.py? 

The value of `account` defined in submodule2 is assigned `None`. I need to use Request module from FastAPI to access subpackage1 path variable.

If I don't declare account in the definition path variable of subpackage2 it doesn't get added in the swagger UI. I'm bit confused here how to acheive this.

### Example Code

---
src/main.py
```python

from fastapi import FastAPI
from subpackage1.views import router as sub_router1

app = FastAPI()

app.include_router(sub_router1)

...
```
---
src/subpackage1/views.py

```python
from fastapi import APIRouter
from subpackage1.subpackage2.views import router as sub_router2
router = APIRouter(prefix="/{account:path}")

router.include_router(sub_router2)

@router.post("/")
def info(account: str):
    print(f"{account = }")
    ...

```

---
src/subpackage1/subpackage2/views.py

```python
from fastapi import APIRouter


router = APIRouter(prefix="/admin")

@router.post("/")
def info(account: str, req: Request):
    print(f"{account = } | {req.path_params.get('account') = }")
    ...
```

---

### Operating System

Linux

### FastAPI Version

0.110.1

### Pydantic Version

2.6.4

### Python Version

3.11.9

**Comment by YuriiMotov** [comment]
> The value of account defined in submodule2 is assigned None. I need to use Request module from FastAPI to access subpackage1 path variable.

I ran your code and I see `account = 'my_path' | req.path_params.get('account') = 'my_path'` when I request POST `http://127.0.0.1:8000/my_path/admin/`.

In swagger docs it also looks Ok.


I think that specifying path parameters in router's `prefix` parameter is undocumented thing (am I right here?) and it shouldn't be used.
It partially work's somehow, but since it's undocumented, nobody guarantees you anything.

**Reply by SDAravind** [reply]
### Case 1: Path Parameter doesn't appear in the nested router in swagger UI

- If you remove account parameter in subpackage2 `views.py`. Path parameter won't appear in swagger.
- I can't even pass account parameter using swagger UI

src/subpackage1/subpackage2/views.py
```python
def create_account_user(
    req: Request
    # account: str):
    print(f"{req.path_params.get('account') = }")
```

![image](https://github.com/tiangolo/fastapi/assets/72403396/1d349d06-7b52-414a-9040-e190894ea8fb)

---

### Case 2: Path Parameter appears in the nested router in swagger UI

- If you add account parameter in subpackage2 `views.py`. I get `None` for account even though I passed demo and I need fetch this value using Request class from FastAPI

src/subpackage1/subpackage2/views.py
```python
def create_account_user(req: Request, account: str):
    print(f"{account = } | {req.path_params.get('account') = }")
    ...
```

![image](https://github.com/tiangolo/fastapi/assets/72403396/db46ebe3-3db3-4694-b015-fc1ba0057e2c)

Output:
![image](https://github.com/tiangolo/fastapi/assets/72403396/68ef0081-80d2-4639-bb86-b0fdb7060776)

**Reply by YuriiMotov** [reply]
I just copied and ran your code with exactly the same versions of FastAPI (0.110.1) and Pydantic (2.6.4) and it works perfectly fine.

I can see `account = 'demo' | req.path_params.get('account') = 'demo'` in the output.

The only difference is that I use Python 3.11.12 instead of 3.11.9.

**Comment by YuriiMotov** [comment]
I'm closing it. If you still have troubles with this problem, feel free to provide more information and reopen it
